pkg/fileutils/fileutils.go

```diff
@@ -7,21 +7,50 @@ import (
 )
 
 // Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
 func Matches(relFilePath string, patterns []string) (bool, error) {
-	for _, exclude := range patterns {
-		matched, err := filepath.Match(exclude, relFilePath)
+
+	matched := false
+
+	for _, pattern := range patterns {
+
+		var negative bool
+		if pattern == "" {
+			continue
+		}
+
+		if pattern[0] == '!' {
+			if len(pattern) == 1 {
+				continue
```

Comment:
Where did we end up on this discussion?  I was thinking we should generate an error instead of silently ignore these lines.